pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -501,6 +501,168 @@ private static String getRepeatedEqualsGeneration(String generatedCodeSoFar, Fie
+		for(Field f : fields) {
+			if (f.parent() != null) {
+				final OneOfField oneOfField = f.parent();
+				generatedCodeSoFar += getFieldsCompareToStatements(oneOfField.fields(), generatedCodeSoFar);
+			}
+
+			if (f.optionalValueType()) {
+				generatedCodeSoFar += getPrimitiveWrapperCompareToGeneration(f);
+			}
+			else if (f.repeated()) {
+				// intentionally skipping repeated fields
+			} else {
+				f.nameCamelFirstLower();
+				if (f.type() == Field.FieldType.FIXED32 ||
+						f.type() == Field.FieldType.INT32 ||
+						f.type() == Field.FieldType.SFIXED32 ||
+						f.type() == Field.FieldType.SINT32 ||
+						f.type() == Field.FieldType.UINT32) {
+					generatedCodeSoFar +=
+							"""
+                            result = Integer.compare($fieldName, thatObj.$fieldName);
+                            if (result != 0) {
+                                return result;
+                            }
+                            """.replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.FIXED64 ||
+						f.type() == Field.FieldType.INT64 ||
+						f.type() == Field.FieldType.SFIXED64 ||
+						f.type() == Field.FieldType.SINT64 ||
+						f.type() == Field.FieldType.UINT64) {
+					generatedCodeSoFar +=
+							"""
+                            result = Long.compare($fieldName, thatObj.$fieldName);
+                            if (result != 0) {
+                                return result;
+                            }
+                            """.replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.BOOL) {
+					generatedCodeSoFar +=
+							"""
+                            result = Boolean.compare($fieldName, thatObj.$fieldName);
+                            if (result != 0) {
+                                return result;
+                            }
+                             """.replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.FLOAT) {
+					generatedCodeSoFar +=
+							"""
+                            result = Float.compare($fieldName, thatObj.$fieldName);
+                            if (result != 0) {
+                                return result;
+                            }
+                             """.replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.DOUBLE) {
+					generatedCodeSoFar +=
+							"""
+                            result = Double.compare($fieldName, thatObj.$fieldName);
+                            if (result != 0) {
+                                return result;
+                            }
+                             """.replace("$fieldName", f.nameCamelFirstLower());
+				} else if(f.type() == Field.FieldType.BYTES) {
+					generatedCodeSoFar +=
+							"""
+							// byte-to-byte comparison is most likely impractical, so we just compare the length
```

Comment:
Even safer, if we cannot support comparison of fields, then do not implement `Comparable` for that message.  Generally most messages should not be `Comparable`, just those that explicitly request to be `Comparable`; and then, ideally, the request specifies the fields and order in which to compare them.
Failing to compile when a message requests to be `Comparable` but has non-comparable fields specified, helps the author make reasonable choices about what should or should not be `Comparable`.

Reply:
@anthony-swirldslabs 
I had doubts about this one, but it looks like it would be more consistent just not support byte array comparison at all. 

@jsync-swirlds 
>Generally most messages should not be Comparable,

I thought the general agreement was to make them all comparable by default and customize the list of fields where needed.

Reply:
That is not a general agreement I am aware of; perhaps the discussion was a bit under-communicated.  Also, it will break a lot of necessary assumptions to implement universally.
I am not confident it is possible to make everything Comparable and be _correct_ in that implementation.  Given that, implementing an incorrect or incomplete comparison method is worse than not implementing anything.  Only implementing Comparable when requested seems, to me, to be a good balance.

The Issue that this PR addresses, and the understanding I have, is that we will implement Comparable *when requested* and then only when a correct and complete implementation is reasonable; throwing a compile error if Comparable is requested but the message includes currently unsupported field types.

Reply:
> The Issue that this PR addresses, and the understanding I have, is that we will implement Comparable when requested and then only when a correct and complete implementation is reasonable; throwing a compile error if Comparable is requested but the message includes currently unsupported field type

This is also my expectation. I think this is safer, and if we find that we can in fact safely generate Comparable for all types in the future, then we can relax this. We really only need a few simple types to be Comparable in services code today, and those we can easily annotate and we can be sure they will be proper implementations.

BTW, the assumption that we can skip byte arrays is probably not true. The Account ID has a one-of third field that can be a byte array, and sorting by Account ID is definitely something we will require.

Reply:
I agree with @jsync-swirlds and @rbair23. In Services, we need only a few keys to be comparable. But if a record is comparable, we must be sure it is correct.

Reply:
Sounds good, I  made the change to the implementation